packages/mdc-chips/chip/foundation.ts

```diff
@@ -94,19 +94,24 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
   }
 
   getDimensions(): ClientRect {
-    const rootRect = this.adapter_.getRootBoundingClientRect();
-    const checkmarkRect = this.adapter_.getCheckmarkBoundingClientRect();
+    const getRootRect = () => this.adapter_.getRootBoundingClientRect();
+    const getCheckmarkRect = () => this.adapter_.getCheckmarkBoundingClientRect();
 
     // When a chip has a checkmark and not a leading icon, the bounding rect changes in size depending on the current
     // size of the checkmark.
-    if (!this.adapter_.hasLeadingIcon() && checkmarkRect !== null) {
-      // The checkmark's width is initially set to 0, so use the checkmark's height as a proxy since the checkmark
-      // should always be square.
-      const width = rootRect.width + checkmarkRect.height;
-      return {...rootRect, width};
-    } else {
-      return rootRect;
+    if (!this.adapter_.hasLeadingIcon()) {
+      const checkmarkRect = getCheckmarkRect();
+      if (checkmarkRect) {
+        const rootRect = getRootRect();
+        const height = rootRect.height;
+        // The checkmark's width is initially set to 0, so use the checkmark's height as a proxy since the checkmark
```

Comment:
I'm confused by what "use the checkmark's height as a proxy" means. Is this comment still relevant?

Reply:
I think it has to do with how the checkmark animates from `width: 0` to `width: auto` (or w/e).

Since the checkmark starts at `width: 0`, we can't query its `width` to figure out how wide it's eventually going to be when it's done animating.

However, we _can_ assume that the checkmark will have square dimensions, so querying its `height` is the same as querying its (future) `width`.

Does that make sense?

Reply:
Well I understand now. I do think the comment could be clarified though.

>The checkmark is a square meaning the client rect's width and height are equivalent. However, the checkbox is initially hidden by setting the width to 0. To account for a potential 0 width checkbox, we use the checkbox's height when adding it to the root client rect's width.

Reply:
Use your comment. That's perfect.